langchain/main.py

```diff
@@ -0,0 +1,46 @@
+import uvicorn
+from fastapi import FastAPI
+from dispatch.fastapi import Dispatch
+
+from dataclasses import dataclass
+from langchain_openai import ChatOpenAI
+from langchain.prompts import ChatPromptTemplate
+
+app = FastAPI()
+dispatch = Dispatch(app)
+
+
+@dataclass
+class TranslateParams:
+    phrase: str
+    language: str
+
+
+@dispatch.function
+def translate_with(params: TranslateParams) -> dict:
+    print(f"Translating '{params.phrase}' to '{params.language}'")
+
+    template = """You are a helpful assistant who translates between languages.
+    Translate the following phrase into the specified language: {phrase}
+    Language: {language}"""
+    chat_prompt = ChatPromptTemplate.from_messages(
+        [
+            ("system", template),
+            ("human", "Translate"),
+        ]
+    )
+    chain = chat_prompt | ChatOpenAI()
+    result = chain.invoke({
+        "phrase": params.phrase,
+        "language": params.language,
+        })
+    print("Translation: ", result.content)
+
+
+@app.post("/translate")
+def translate(phrase: str, language: str):
+    translate_with.dispatch(TranslateParams(phrase=phrase, language=language))
+
+
+if __name__ == "__main__":
+    uvicorn.run(app, host="localhost", port=8000)
```

Comment:
why not use `dispatch.run` here instead of a FastAPI server?

Reply:
Do we have the facilities to expose specific routes (here /translate) with run?

Reply:
It's a different use case I guess, the http endpoint with FastAPI doesn't do much but give a mechanism to trigger the workflow, my thoughts were that removing that and just adding a .dispatch call in the example could keep it simpler.

Reply:
I prefer keeping the path right now to illustrate this example. We will have more examples showing `dispatch.run`

Reply:
OK, to be clear, it's not that I thought it would be better to showcase `dispatch.run`, but that we could make the example simpler to run. When triggered via HTTP, there's an extra step to run the code.